src/test/java/com/github/fishio/TestPowerUpSpawner.java

```diff
@@ -0,0 +1,155 @@
+package com.github.fishio;
+
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.never;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Mockito.times;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.github.fishio.power_ups.PowerUp;
+import com.github.fishio.power_ups.PuExtraLife;
+import com.github.fishio.power_ups.PuFreeze;
+import com.github.fishio.power_ups.PuSuperSpeed;
+
+/**
+ * Tests the PowerUpSpawer class.
+ */
+public class TestPowerUpSpawner {
+
+	private PowerUpSpawner pus;
+	
+	private PlayingField pf;
+	
+	private Random rand;
+	
+	/**
+	 * Sets the PowerUpSpawner attribute in this class
+	 * before every test.
+	 */
+	@Before
+	public void setUp() {
+		this.pf = Mockito.mock(PlayingField.class);
+		when(pf.getFPS()).thenReturn(60); //Making sure our PowerUpSpawner doesn't think the FPS is 0.
+		when(pf.getWidth()).thenReturn(100); //Same as above
+		
+		this.pus = Mockito.spy(new PowerUpSpawner(pf));
+		
+		this.rand = Mockito.mock(Random.class);
+		this.pus.setRandom(rand);
+	}
+	
+	/**
+	 * Tests the first case in the switch statement of the
+	 * getRandomPowerUp method.
+	 */
+	@Test
+	public void testGetRandomPowerUp1() {
+		when(rand.nextInt(pus.getPowerUpCount())).thenReturn(0);
+		
+		PowerUp pu = pus.getRandomPowerUp();
+		assertTrue(pu instanceof PuFreeze);
+		
+		//TODO check the correct CollisionMask of the PowerUp
+		//once CollisionMasks are implemented in PowerUps.
+	}
+	
+	/**
+	 * Tests the second case in the switch statement of the
+	 * getRandomPowerUp method.
+	 */
+	@Test
+	public void testGetRandomPowerUp2() {
+		when(rand.nextInt(pus.getPowerUpCount())).thenReturn(1);
+		
+		PowerUp pu = pus.getRandomPowerUp();
+		assertTrue(pu instanceof PuSuperSpeed);
+		
+		//TODO check the correct CollisionMask of the PowerUp
+		//once CollisionMasks are implemented in PowerUps.
+	}
+	
+	/**
+	 * Tests the third case in the switch statement of the
+	 * getRandomPowerUp method.
+	 */
+	@Test
+	public void testGetRandomPowerUp3() {
+		when(rand.nextInt(pus.getPowerUpCount())).thenReturn(2);
+		
+		PowerUp pu = pus.getRandomPowerUp();
+		assertTrue(pu instanceof PuExtraLife);
+	}
+	
+	/**
+	 * Tests the default case in the switch statement of the
+	 * getRandomPowerUp method.
+	 */
+	@Test
+	public void testGetRandomPowerUpDefault() {
+		int defaultIndex = pus.getPowerUpCount();
+		when(rand.nextInt(pus.getPowerUpCount())).thenReturn(defaultIndex);
+		
+		PowerUp pu = pus.getRandomPowerUp();
+		assertTrue(pu == null);
```

Comment:
*Auto-generated comment*

    
    Use assertNull(x) instead of assertTrue(x==null), or assertNotNull(x) vs assertFalse(x==null)